src/abstractdataframe/show.jl

```diff
@@ -55,22 +55,48 @@ end
 
 """Return compact string representation of type T"""
 function compacttype(T::Type, maxwidth::Int=8)
+    maxwidth = max(8, maxwidth)
+
     T === Any && return "Any"
     T === Missing && return "Missing"
+
     sT = string(T)
-    length(sT) ≤ maxwidth && return sT
+    textwidth(sT) ≤ maxwidth && return sT
+
     if T >: Missing
         T = nonmissingtype(T)
         sT = string(T)
-        suffix = "⍰"
-        # handle the case when after removing Missing union type name is short
-        length(sT) ≤ 8 && return sT * suffix
+        suffix = "?"
+        maxwidth -= 1 # we will add "?" at the end
+        textwidth(sT) ≤ maxwidth && return sT * suffix
     else
         suffix = ""
     end
-    T <: Union{CategoricalString, CategoricalValue} && return "Categorical…"*suffix
-    # we abbreviate consistently to total of 8 characters
-    match(Regex("^.\\w{0,$(7-length(suffix))}"), sT).match * "…"*suffix
+
+    maxwidth -= 1 # we will add "…" at the end
+
+    if T <: Union{CategoricalString, CategoricalValue}
+        return (maxwidth ≥ 11 ? "Categorical…" : "Cat…") * suffix
```

Comment:
If the column is wider than 17 characters, can we print `CategoricalValue`/`CategoricalString` in full? For simplicity, we can use the same threshold for both, anyway the latter is going away soon.

Reply:
Currently there is one nuance that `T` can also be `Union{CategoricalString, CategoricalValue}` (probably a very improbable case). I will fix it then and we can do what you propose.